### PR TITLE
feat: add advanced font config

### PR DESCRIPTION
**Pull request by doinkythederp**
## Describe the changes this PR makes. Why should it be merged?

Adds advanced font configuration so that users can specify an exact size and change the font family of their text.

## Additional Context
- I have tested these changes on a VEX V5 Brain.
- These are breaking changes (semver: major).
- These changes update the crate's interface (e.g. functions/modules added or changed).

### CODE REVIEW

**Review by Gavin-Niederman — APPROVED** [review]
LGTM

File: armv7a-vex-v5.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "panic-strategy": "abort",
     "relocation-model": "static",
     "llvm-target": "armv7a-none-eabihf",
-    "features": "+v7,+neon,+vfp3,+thumb2",
+    "features": "+v7,+neon,+vfp3d16,+thumb2",
```
Comment (zabackary) [inline armv7a-vex-v5.json:11]: Was this intentional? It looks like [the tip of the tree](https://github.com/vexide/cargo-v5/blob/main/armv7a-vex-v5.json) from `cargo-v5` does not include this change.

Comment (doinkythederp) [reply armv7a-vex-v5.json:11]: This change is from <https://github.com/vexide/cargo-v5/blob/main/src/targets/armv7a-vex-v5.json>

Comment (Gavin-Niederman) [reply armv7a-vex-v5.json:11]: Just added this change to cargo-v5.

File: packages/vexide-devices/src/display.rs [hunk 2]
```diff
@@ -253,14 +253,169 @@ impl Fill for Rect {
...
+    }
+
+    /// An extra-small font size with a value of one-fifth.
+    pub const EXTRA_SMALL: Self = Self::new(1, 5);
+    /// A small font size with a value of one-fourth.
+    pub const SMALL: Self = Self::new(1, 4);
+    /// A medium font size with a value of one-third.
+    pub const MEDIUM: Self = Self::new(1, 3);
+    /// A medium font size with a value of one-half.
+    pub const LARGE: Self = Self::new(1, 2);
```
Comment (doinkythederp) [inline packages/vexide-devices/src/display.rs:370]: comment issue here

File: packages/vexide-devices/src/display.rs [hunk 3]
```diff
@@ -253,14 +253,169 @@ impl Fill for Rect {
...
+    ///
+    /// This function is lossy, but negligibly so.
+    /// The highest the denominator can be is 10000.
+    ///
+    /// # Errors
+    ///
+    /// - [`NegativeFontSizeError`] if the given size is negative.
+    pub fn from_float(size: f32) -> Result<Self, NegativeFontSizeError> {
+        if size.is_sign_negative() {
+            return Err(NegativeFontSizeError { value: size });
```
Comment (doinkythederp) [inline packages/vexide-devices/src/display.rs:349]: may want to use `expect!()` here

File: packages/vexide-devices/src/display.rs [hunk 4]
```diff
@@ -253,14 +253,169 @@ impl Fill for Rect {
...
+        Self::from_float(value)
+    }
+}
+impl TryFrom<f64> for FontSize {
+    type Error = NegativeFontSizeError;
+
+    fn try_from(value: f64) -> Result<Self, Self::Error> {
+        Self::from_float(value as f32)
+    }
+}
```
Comment (doinkythederp) [inline packages/vexide-devices/src/display.rs:395]: maybe also want to impl `From<u32>`

File: packages/vexide-devices/src/display.rs [hunk 5]
```diff
@@ -336,20 +491,7 @@ impl Text {
...
-                }
-                TextSize::Medium => {
-                    vexDisplayStringAt(0, 0, c"".as_ptr());
-                }
-                TextSize::Large => {
-                    vexDisplayBigStringAt(0, 0, c"".as_ptr());
-                }
-            }
-
+            self.font.apply();
```
Comment (doinkythederp) [inline packages/vexide-devices/src/display.rs:494]: want to test if this works as a replacement before merging

**Review by doinkythederp — COMMENTED** [review]
Things I want to look at before merging

File: packages/vexide-devices/src/display.rs (hunk 3 above)
Comment (zabackary) [reply packages/vexide-devices/src/display.rs]: `ensure!()`?

File: packages/vexide-devices/src/display.rs (hunk 5 above)
Comment (zabackary) [reply packages/vexide-devices/src/display.rs:503]: Tested and seems to work

**Pull request merged**